pkg/container/batch/tools.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package batch
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/pb/api"
+)
+
+func BatchToProtoBatch(bat *Batch) (*api.Batch, error) {
```

Comment:
What is the difference between two batch and why can't we use one batch definition?

Reply:
> I wonder if you can try to get rid of TAE vector.
> 
> WHY do we need so many of that? Also, why do we need a proto vector? Isn't vector serialized and deserialized to bytes -- that is all?

`Proto vector` had been involved in before this PR .